amd/src/carousel.js

```diff
@@ -22,17 +22,29 @@
  */
 define(['jquery', 'core/modal_factory', 'core/ajax', 'block_carousel/slick'], function($, ModalFactory, Ajax) {
     return {
-        init: function(blockid, playspeed) {
+        init: function(blockid, numslides, autoplay, playspeed) {
             $('#carousel' + blockid + ' .slidewrap').show();
-            $('#carousel' + blockid).slick({
+            var carousel = $('#carousel' + blockid);
+            carousel.slick({
                 dots: true,
                 infinite: true,
                 speed: 300,
-                slidesToShow: 1,
+                slidesToShow: numslides,
                 adaptiveHeight: true,
-                autoplay: true,
-                autoplaySpeed: playspeed
+                autoplay: autoplay,
+                autoplaySpeed: playspeed,
+                focus
+            });
+
+            // This is a special case for when carousel is embedded in another block.
+            // When the carousel is resized, we need to resize.
+            // This causes slick to natively resize itself.
+            const observer = new ResizeObserver(function() {
+                $(window).trigger('resize');
+                $(document).trigger('ready');
```

Comment:
ready seems a bit heavy, what is on on the other side listening for this? seems like it could have a lot of side effects on other random stuff. There might be a more narrowly focused event we can send which does the same job